### PR TITLE
Add functional tests for reconciliation endpoint Authorization

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1360

### Change description ###
Add functional tests for Reconciliation endpoint authorization. 
Scenarios:
- Authorization header with **valid** API Key (should return 200)
- Authorization header with **invalid** API key (should return 401)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/ReconciliationControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        String report = Resources.toString(
+            getResource("reconciliation/reconciliation-supplier-report.json"),
+            UTF_8
+        );
+
+        RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(config.blobRouterUrl)
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/ReconciliationControllerTest.java:51]: can we also have synthetic header as well?

**Review by doncem — APPROVED** [review]

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -31,6 +31,7 @@ def combinedSecrets = [
 
      secret('storage-account-staging-secondary-key', 'TEST_STORAGE_ACCOUNT_KEY'),
      secret('storage-account-staging-name', 'TEST_STORAGE_ACCOUNT_NAME'),
+     secret('reconciliation-api-key', 'TEST_RECONCILIATION_API_KEY'),
```
Comment (doncem) [inline Jenkinsfile_CNP:34]: probably don't need that extra comma there ;)

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/ReconciliationControllerTest.java [hunk 3]
```diff
@@ -49,9 +50,10 @@ private void postReconciliationSupplierReport(String authKey, int expectedStatus
             .relaxedHTTPSValidation()
             .baseUri(config.blobRouterUrl)
             .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
-            .body(report)
+            .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Blob Router Service Functional test")
             .header(HttpHeaders.AUTHORIZATION, authKey)
             .body(report)
+            .body(report)
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/ReconciliationControllerTest.java:56]: oops

**Review by doncem — APPROVED** [review]

File: src/functionalTest/resources/application.conf [hunk 4]
```diff
@@ -1,5 +1,7 @@
 blob-router-url=${TEST_URL}
 
+reconciliation-api-key=${TEST_RECONCILIATION_API_KEY}
```
[the remark this answers is not in the document]
Comment (aliveni) [reply src/functionalTest/resources/application.conf:3]: raised https://github.com/hmcts/cnp-flux-config/pull/5597

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/ReconciliationControllerTest.java (hunk 3 above)
Comment (aliveni) [reply src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/ReconciliationControllerTest.java]: removed 🙈

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**